Content.Client/GameObjects/Components/FoldableVisualizer.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Linq;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.GameObjects.Components.Strap;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Client.GameObjects.Components
+{
+
+    /// <summary>
+    /// Updates the visual state for <see cref="SharedFoldableComponent"/>
+    /// </summary>
+    [UsedImplicitly]
+    public class FoldableVisualizer : AppearanceVisualizer
+    {
+        /// <summary>
+        /// The suffix used to define the buckled states names in RSI file.
+        /// [state] becomes [state][suffix],
+        /// default suffix is _buckled, so chair becomes chair_buckled
+        /// </summary>
+        [ViewVariables][DataField("foldedSuffix", required: false)]
+        private string? _foldedSuffix = "_folded";
+
+        // This keeps a reference to the entity's SpriteComponent to avoid fetching it each time
+        private ISpriteComponent? _sprite;
+
+        private bool? _isFolded;
```

Comment:
These are flyweighted so you can't store any entity specific data in visualizers.